Shrink our gigantic tests!

Wafflespeanut opened the pull request:
This makes use of a simple wrapper around the JSON object and marks stuff whenever we 'get' something from the dictionary (using `[]`). This can help shrink the gigantic tests we currently have (which has a hell lot of unused key/val)

highfive commented:
Heads up! This PR modifies the following files:
 * @jdm: helpers.py, eventhandler.py, test.py, json_cleanup.py

Wafflespeanut commented:
gotta fix the flake8 stuff...

Wafflespeanut commented:
It might be worth taking a look at the failed Travis build! :smile:

Wafflespeanut commented:
@jdm I think it's still buggy, but well, this is just a prototype. I'm curious whether this would be useful if I make it to work.

highfive commented:
New code was committed to pull request.

Wafflespeanut commented:
Now, [this](https://travis-ci.org/servo/highfive/builds/128450660) is progress! (still, 3 tests fail)

highfive commented:
New code was committed to pull request.

Wafflespeanut commented:
All the tests pass now, but I had to put some type conversions to integrate the lint.

Wafflespeanut commented:
After trying to integrate this into highfive (for a while now), I realize that the entire problem lies on creating the tests, where we read the JSON files twice (one for the initial/expected stuff), and the other for payload. I think I should modify the linter to take objects instead of going directly for the file.

highfive commented:
New code was committed to pull request.

Wafflespeanut commented:
- fixed the linter for flagging only the stuff in `payload`
- avoided reading the test JSONs twice
- added flag to warn/overwrite the nodes in test files

only the command-line argument is left to go...

highfive commented:
New code was committed to pull request.

Wafflespeanut commented:
I'm finally satisfied with [the output](https://travis-ci.org/servo/highfive/builds/128686632) :smile:

bors-servo commented:
:umbrella: The latest upstream changes (presumably #118) made this pull request unmergeable. Please resolve the merge conflicts.

highfive commented:
New code was committed to pull request.

highfive commented:
New code was committed to pull request.

highfive commented:
New code was committed to pull request.

highfive commented:
New code was committed to pull request.

highfive commented:
New code was committed to pull request.

Wafflespeanut commented:
This is ready for review!

highfive commented:
New code was committed to pull request.

highfive commented:
New code was committed to pull request.

jdm commented on handlers/assign_reviewer/__init__.py:
```diff
@@ -55,7 +55,7 @@ def on_pr_opened(self, api, payload):
         # If the pull request already has an assignee,
         # don't try to set one ourselves.
-        if pr["assignee"] != None:
+        if pr["assignee"] != None:      # NOQA (silence flake8 here)
```
What is it complaining about?

Wafflespeanut replied:
flake8 prefers `is not None`, but we need `!=` to call our overridden `__ne__` method (the former checks for an exact match of the object, which will fail while testing)

jdm commented:
The need for `str(foo)` and `int(foo)` in various places is disappoint. Is there no way around that?

highfive commented:
New code was committed to pull request.

Wafflespeanut commented:
True, they *are* bad, but I couldn't find a way around them. I've fixed some of them, and now we have three situations (in regex matching) where `str(foo)` occurs. Python's `re` methods look pretty strict, that they accept only string/unicode, which prevents us from overriding anything :disappointed:

jdm commented:
Thanks!

Wafflespeanut commented:
Thanks for reviewing it :)
